File: designpatterns/src/com/renho/behavior/state/ConcreteStateC.java
package com.renho.behavior.state;

public class ConcreteStateC extends State{

	private int status;
	
	public ConcreteStateC(int status) {
		this.status = status;
	}
	
	@Override
	public void handle(Context context) {
		if(this.status >= 24) {
			System.out.println("已经休息:" + status);	
		}else {
			System.out.println("晚上:" + status);			
			status = status + 6;
		}
	}

	public int getStatus() {
		return status;
	}

	public void setStatus(int status) {
		this.status = status;
	}

}

File: ideaworkspace/designpatterns/src/main/scala/com/renho/behavior/state/ConcreteStateC.scala
package com.renho.behavior.state

/**
  * Created by renho on 2016/7/11.
  */
class ConcreteStateC(var status: Int) extends State {
	override def handle(context: Context): Unit = {
		if(this.status >= 24) {
			println("已经休息:" + status);
		}else {
			println("晚上:" + status);
			status = status + 6;
		}
	}
}
